internal/fields/sql.go

```diff
@@ -19,6 +19,18 @@ type Valuer struct {
 	value reflect.Value
 }
 
+var marshalerType = reflect.TypeOf((*marshaler)(nil)).Elem()
+
+func nonPointerMarshal(d *Descriptor, val reflect.Value) (reflect.Value, bool) {
```

Comment:
TODO: add a comment here explaining exactly what this is doing

Reply:
+1
Naming the return types might help too (especially the bool)

Reply:
👍